stdlib/LinearAlgebra/src/dense.jl

```diff
@@ -1568,4 +1568,4 @@ function lyap(A::StridedMatrix{T}, C::StridedMatrix{T}) where {T<:BlasFloat}
     rmul!(Q*(Y * adjoint(Q)), inv(scale))
 end
 lyap(A::StridedMatrix{T}, C::StridedMatrix{T}) where {T<:Integer} = lyap(float(A), float(C))
-lyap(a::T, c::T) where {T<:Number} = -c/(2a)
+lyap(a::T, c::T) where {T<:Number} = -c/(2real(a))
```

Comment:
```suggestion
lyap(a::Union{Real,Complex}, c::Union{Real,Complex}) = -c/(2real(a))
```

Reply:
@dkarrasch I have committed the suggestion above. Should I add a separate definition for handling `Quaternion`s? Should I create a separate issue for this?

Reply:
No, I don't think we need an issue. Let's wait until somebody hits the missing method error. This has been wrong even in the complex case, not to mention the quaternion case (if that has ever been used).